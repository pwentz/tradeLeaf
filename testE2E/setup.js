import React, { Component } from 'react';
const { View, StyleSheet } = require('react-native');

import { Provider } from 'react-redux';

import MockApi from './mocks/MockApi';
const { setupApp } = require('../setupApp');

__DEV__ = true;

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
    canOpenURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
    getInitialURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
  };
});

jest.mock('ScrollView', () => jest.genMockFromModule('ScrollView'));

export const renderer = require('react-test-renderer');

const AccountRequirements = require('../src/components/register/AccountRequirements').default;
const App = require('../src/components/app/App').default;
const Avatar = require('../src/components/common/Avatar').default;
const Card = require('../src/components/matchboard/Card').default;
const LoginForm = require('../src/components/login/LoginForm').default;
const RegisterForm = require('../src/components/register/RegisterForm').default;
const TabBar = require('../src/components/app/TabBar').default;
const ChatPreview = require('../src/components/chat/ChatPreview').default;

export const components = {
  LoginForm,
  RegisterForm,
  Card,
  AccountRequirements,
  Avatar,
  TabBar,
  ChatPreview,
};

class TestRouteObserver {}

export function setupTestApp() {
  const api = new MockApi();

  const shouldLog = false;

  const { actions, store } = setupApp(api, shouldLog);

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'stretch',
      backgroundColor: 'white',
    },
  });

  const testRouteObserver = new TestRouteObserver();

  class TradeLeafApp extends Component {
    render() {
      return (
        <View style={styles.container}>
          <Provider store={store}>
            <App screenProps={{ actions, testRouteObserver }} />
          </Provider>
        </View>
      );
    }
  }

  return {
    TradeLeafApp,
    api,
    actions,
    dispatch: store.dispatch,
    testRouteObserver,
  };
}
