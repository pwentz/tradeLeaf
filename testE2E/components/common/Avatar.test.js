import React from 'react';

import { components, renderer } from '../../setup';

const noop = () => {}

it('<Avatar />', () => {
  const Avatar = components.Avatar;
  const component = renderer.create(
    <Avatar
      onPressEdit={noop}
    />
  );

  expect(component.toJSON()).toMatchSnapshot();
})

it('<Avatar onPressEdit={null} />', () => {
  const Avatar = components.Avatar;
  const component = renderer.create(
    <Avatar />
  );

  expect(component.toJSON()).toMatchSnapshot();
})
