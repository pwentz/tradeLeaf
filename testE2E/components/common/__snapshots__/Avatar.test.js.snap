// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "position": "relative",
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "borderColor": "grey",
            "borderWidth": 1,
          },
          Object {
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "left": 52.53940455341506,
          "position": "absolute",
          "top": 61.403912543153055,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "backgroundColor": "#27496B",
              "borderRadius": 80,
              "overflow": "hidden",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 7.2727272727272725,
                },
                Object {
                  "margin": 4,
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<Avatar onPressEdit={null} /> 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "borderColor": "grey",
          "borderWidth": 1,
        },
        Object {
          "borderRadius": 20,
          "height": 40,
          "width": 40,
        },
      ]
    }
  />
</View>
`;
